core/src/main/scala/cats/data/Streaming.scala

```diff
@@ -523,16 +524,17 @@ sealed abstract class Streaming[A] extends Product with Serializable { lhs =>
    * If no elements satisfy `f`, the current stream will be returned.
    *
    * For example:
-   *
-   *   Streaming(1, 2, 3, 4, 5, 6, 7).takeWhile(n => n != 4)
-   *
-   * Will result in: Streaming(4, 5, 6, 7)
+   * {{{
+   * scala> val s = Streaming(1, 2, 3, 4, 5, 6, 7)
+   * scala> s.dropWhile(n => n != 4).toList
+   * res0: List[Int] = List(4, 5, 6, 7)
+   * }}}
    */
   def dropWhile(f: A => Boolean): Streaming[A] =
     this match {
       case Empty() => Empty()
       case Wait(lt) => Wait(lt.map(_.dropWhile(f)))
-      case Cons(a, lt) => if (f(a)) Empty() else Cons(a, lt.map(_.dropWhile(f)))
+      case Cons(a, lt) => if (f(a)) Wait(lt.map(_.dropWhile(f))) else Cons(a, lt)
```

Comment:
Cool - that makes a lot of sense - and thanks a million for following up!!! 

I guess in this instance there would have been no value, as there is no additional type information required, but I can definitely imagine scenarios, where this could be valuable.